### ISSUE TITLE
FindSodium.cmake doesn't find shared libsodium

### ISSUE THREAD

**Issue by trunc88** [Syndace/libxeddsa#3]
FindSodium.cmake doesn't find the libsodium shared library on my Linux system. I was able to get it to work by commenting out some of the static library detection like so:
```
--- FindSodium.cmake.orig	2022-11-29 22:22:39.845951870 -0500
+++ FindSodium.cmake.worksmin	2022-11-29 22:51:52.737827922 -0500
@@ -80,28 +80,28 @@
         pkg_check_modules(sodium_PKG QUIET libsodium)
     endif()
 
-    if(sodium_USE_STATIC_LIBS)
-        foreach(_libname ${sodium_PKG_STATIC_LIBRARIES})
-            if (NOT _libname MATCHES "^lib.*\\.a$") # ignore strings already ending with .a
-                list(INSERT sodium_PKG_STATIC_LIBRARIES 0 "lib${_libname}.a")
-            endif()
-        endforeach()
-        list(REMOVE_DUPLICATES sodium_PKG_STATIC_LIBRARIES)
+#    if(sodium_USE_STATIC_LIBS)
+#        foreach(_libname ${sodium_PKG_STATIC_LIBRARIES})
+#            if (NOT _libname MATCHES "^lib.*\\.a$") # ignore strings already ending with .a
+#                list(INSERT sodium_PKG_STATIC_LIBRARIES 0 "lib${_libname}.a")
+#            endif()
+#        endforeach()
+#        list(REMOVE_DUPLICATES sodium_PKG_STATIC_LIBRARIES)
 
         # if pkgconfig for libsodium doesn't provide
         # static lib info, then override PKG_STATIC here..
-        if (NOT sodium_PKG_STATIC_FOUND)
-            set(sodium_PKG_STATIC_LIBRARIES libsodium.a)
-        endif()
+#        if (NOT sodium_PKG_STATIC_FOUND)
+#            set(sodium_PKG_STATIC_LIBRARIES libsodium.a)
+#        endif()
 
-        set(XPREFIX sodium_PKG_STATIC)
-    else()
+#        set(XPREFIX sodium_PKG_STATIC)
+#    else()
         if (NOT sodium_PKG_FOUND)
             set(sodium_PKG_LIBRARIES sodium)
         endif()
 
         set(XPREFIX sodium_PKG)
-    endif()
+#    endif()
 
     find_path(sodium_INCLUDE_DIR sodium.h
         HINTS ${${XPREFIX}_INCLUDE_DIRS}
```
After these changes, it finds libsodium.so and compiles without issue.

Thank you

**Comment by fantomfp** [comment]
Which distro are you using?

**Comment by Syndace** [comment]
The build expects you to have both the static and dynamic versions of libsodium installed. If you had to comment out the part that looks for the static version, you likely didn't have it installed. I will look into updating either the build scripts or the documentation.

**Comment by Syndace** [comment]
Thanks for the report. This was fixed with 8ab957a, which removes the dependency on the static version of libsodium.

**Comment by trunc88** [comment]
@fantomfp This is on Obarun, libsodium is located at /usr/lib/libsodium.so.23.3.0, with symlinks from libsodium.so.23 and libsodium.so.

@Syndace OK that makes sense, I don't have the static version installed. Didn't realize it was expecting both to be present. Thank you for the fix. I tested it, and it works for me without modification now.
